synthtool/gcp/gapic_bazel.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Optional, Union
+import os
+import tempfile
+
+from synthtool import _tracked_paths
+from synthtool import log
+from synthtool import metadata
+from synthtool import shell
+from synthtool.sources import git
+
+GOOGLEAPIS_URL: str = git.make_repo_clone_url("googleapis/googleapis")
+GOOGLEAPIS_PRIVATE_URL: str = git.make_repo_clone_url("googleapis/googleapis-private")
+LOCAL_GOOGLEAPIS: Optional[str] = os.environ.get("SYNTHTOOL_GOOGLEAPIS")
+
+
+class GAPICBazel:
+    """A synthtool component that can produce libraries using bazel build.
+    """
+
+    def __init__(self):
+        self._ensure_dependencies_installed()
+        self._googleapis = None
+        self._googleapis_private = None
+
+    def py_library(self, service: str, version: str, **kwargs) -> Path:
```

Comment:
Using kwargs like this makes it harder for me to maintain because my IDE can't show me the list of arguments as I'm typing the function.